### ISSUE TITLE
Array custom field allows Items to be added but not removed.

### ISSUE THREAD

**Issue by aidankilleen** [tabalinas/jsgrid#962]
I have created a custom field that allows the user to edit an array. Initially this is a text box and 
insertTemplate converts the array to a string using .join(v)
the insertValue and editValue functions parse the text back into an array using v.split(",").

When I add an element to the textbox it is stored correctly.

When I try to remove an element from the textbox the array is not updated.

I have created a [jsfiddle]([https://jsfiddle.net/ky4u13of/1/]) that demonstrates this.  You can edit the Colour field to add an element but you cannot remove one.

**Comment by aidankilleen** [comment]
I have tracked down the problem and it is in jsgrid.js line 1292

The _updateRow function is taking the original object and the edited object and merging them together using $.extend.  For normal fields this works fine, the old values are replaced with the new values.  For an array however it merges the two arrays together - causing the reported behavious where you can add items to the array but you cannot remove them.

My temporary fix is to remove any array elements from the original object before calling $.extend:

`       
 _updateRow: function($updatingRow, editedItem) {
            var updatingItem = $updatingRow.data(JSGRID_ROW_DATA_KEY);
            
            // need to remove any arrays from updatingItem
            // $.extend will merge arrays rather than replace them
            // effectively meaning you can't remove items from a property 
            // that is an array.
            for (prop in updatingItem) {
                if (Array.isArray(updatingItem[prop])) {
                    updatingItem[prop] = [];
                }
            }
            var updatingItemIndex = this._itemIndex(updatingItem);
            var updatedItem = $.extend(true, {}, updatingItem, editedItem);
`

**Comment by aidankilleen** [comment]
I would appreciate it if you could confirm that this fix is valid and doesn't have any consequences that I might have missed.

Thanks

**Comment by tabalinas** [comment]
Yes, that's a tricky case. Actually, if we would do non-deep coping everything works fine. But we need it to support nested fields. 

Your hotfix looks valid. While I don't see a proper fix for the moment w/o a breaking change.

**Comment by robertsLando** [comment]
Same problem here, I have fixed this by set an id on items and change the value of arrays in `updateItem` function. Any other ideas?

**Comment by davehayes** [comment]
I just got bit by this too. In my case this behavior of extend causes an inability to delete elements from an array. Would it be possible to add something like this...

```javascript
var updatedItem = this._mergeObjects(updatingItem, editedItem);
...
_mergeObjects: function(ob1, ob2) {
    return $.extend(true, {}, ob1, ob2);
}
```
...? Then we API users can replace ```_mergeObjects``` with our own code. (As it is, I'm going to have to replace ```_updateRow```.)
